execution_engine/src/core/tracking_copy/mod.rs

```diff
@@ -410,9 +428,14 @@ impl<R: StateReader<Key, StoredValue>> TrackingCopy<R> {
         let mut proofs = Vec::new();
 
         loop {
+            if query.depth > config.query_depth_limit {
```

Comment:
I think this lets the query exceed the limit by 1.  For example, it the limit is 1, this will only cause us to return a result on the third iteration of the loop, i.e. after two iterations have completed.

Maybe that's the intent, but it seems a bit non-intuitive to me.  If we want to keep this behaviour, I think we should document it more thoroughly - particularly in more user-facing places like the config file.

Reply:
Fair point. I fixed the looping so it will hard stop exactly at `query.depth >= config.query_depth_limit` and also renamed the config name